lib/transactional.pm

```diff
@@ -176,3 +177,28 @@ sub ensure_rollback_service_not_running {
         $output =~ '^(start|running)$' ? sleep 10 : last;
     }
 }
+
+
+# Verify that essential snapshots cannot be deleted, see https://jira.suse.com/browse/SLE-3804
+sub check_undel_snaps {
+    # Create some snapshots with a transactional update snapshot in the middle.
+    record_info "Check undel", "Check that essential snapshots cannot be deleted, creates snapshots and installs a package";
+    assert_script_run("snapper create");
+    trup_call "ptf install" . rpmver('security');
+    assert_script_run("snapper create");
+
+    my $last_snapshot        = +(split /\n/, script_output('snapper list --disable-used-space'))[-1];
+    my $last_snapshot_number = +(split ' ',  $last_snapshot)[0];
+    my @expected        = ('current system', 'currently mounted', 'to be mounted');
+    my @delete_response = (split /\n/,       script_output("snapper delete 0-$last_snapshot_number 2>&1"));
+
+    for my $line (0 .. $#expected) {
+        record_info("$expected[$line]", "Snapper delete response: $delete_response[$line]");
+        unless ($delete_response[$line] =~ /^Cannot\sdelete.*$expected[$line].*/) {
```

Comment:
I would say it's a matter of taste, I would not find it more readable with !~ and if

Reply:
`unless` is recommended to be used when is clear, combined with a regex doesn't fit that. But you can keep it, it is just for you to know in case other could have the same impression.